src/lib.rs

```diff
@@ -1616,14 +1676,150 @@ mod test {
         assert!(!_NEG1_2.is_integer());
     }
 
+    #[cfg(not(feature = "std"))]
+    use core::fmt;
+    #[cfg(not(feature = "std"))]
+    use fmt::Write;
+    #[cfg(not(feature = "std"))]
+    const TESTER_BUF_SIZE: usize = 32;
+    #[cfg(not(feature = "std"))]
+    #[derive(Debug)]
+    struct NoStdTester {
+        cursor: usize,
+        buf: [u8; TESTER_BUF_SIZE],
+    }
```

Comment:
I hacked this struct together to allow testing in `no_std`, but I suspect there is a better way, that requires less code. I can't use `format!`, because `String`s are not allowed in `no_std`.

Reply:
Hmm, yeah, you basically want `std::io::Cursor`, but for `core::fmt`. I think it's an okay hack for testing.